plugins/rest/rest.go

```diff
@@ -159,11 +162,35 @@ func New(config []byte, opts ...func(*Client)) (Client, error) {
 		return Client{}, err
 	}
 
+	// Ensure we use a http.Transport with proper settings: the zero values are not
```

Comment:
Could you adjust this comment slightly to specify _why_ the default settings are not a good choice? I looked at the linked issue and I _think_ the answer is that it leads to leaks but I'm not sure.